instrumentation/spring/spring-webmvc-3.1/library/src/main/java/io/opentelemetry/instrumentation/spring/webmvc/SpringWebMvcHttpAttributesExtractor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.webmvc;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpServerAttributesExtractor;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+final class SpringWebMvcHttpAttributesExtractor
+    extends HttpServerAttributesExtractor<HttpServletRequest, HttpServletResponse> {
+  @Override
+  protected @Nullable String method(HttpServletRequest request) {
+    return request.getMethod();
+  }
+
+  @Override
+  protected @Nullable String userAgent(HttpServletRequest request) {
+    return request.getHeader("user-agent");
+  }
+
+  @Override
+  protected @Nullable Long requestContentLength(
+      HttpServletRequest request, @Nullable HttpServletResponse response) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable Long requestContentLengthUncompressed(
+      HttpServletRequest request, @Nullable HttpServletResponse response) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable String flavor(
+      HttpServletRequest request, @Nullable HttpServletResponse response) {
+    return request.getProtocol();
+  }
+
+  @Override
+  protected Integer statusCode(HttpServletRequest request, HttpServletResponse response) {
+    return response.getStatus();
```

Comment:
Would it make sense to mimic servlet instrumentation and set status to 500 when response isn't committed and there is an exception?

Reply:
Can you point me to the class where it happens?

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d55eee2ebf74b21ba9f63da44d511d7316d43de8/instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletHttpAttributesExtractor.java#L112
To pass throwable response is wrapped, which happens here
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d55eee2ebf74b21ba9f63da44d511d7316d43de8/instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletHelper.java#L51

Reply:
Nice catch, I'll add this. Thanks!